### ISSUE TITLE
Applying a second azurerm_maintenance_assignment_virtual_machine will fail

### ISSUE THREAD

**Issue by rob31231232** [hashicorp/terraform-provider-azurerm#20397]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.7

### AzureRM Provider Version

3.42.0

### Affected Resource(s)/Data Source(s)

azurerm_maintenance_assignment_virtual_machine

### Terraform Configuration Files

```hcl
resource "azurerm_maintenance_assignment_virtual_machine" "patch_window" {
  location                     = azurerm_resource_group.resource_group.location
  maintenance_configuration_id = data.azurerm_maintenance_configuration.patch_window.id
  virtual_machine_id           = azurerm_windows_virtual_machine.virtual_machine.id
}

resource "azurerm_maintenance_assignment_virtual_machine" "definition_updates" {
  location                     = azurerm_resource_group.resource_group.location
  maintenance_configuration_id = data.azurerm_maintenance_configuration.definition_updates.id
  virtual_machine_id           = azurerm_windows_virtual_machine.virtual_machine.id
}
```

### Debug Output/Panic Output

```shell
Error: A resource with the ID "/subscriptions/<Subscription ID>/resourceGroups/<resource group>/providers/Microsoft.Maintenance/maintenanceConfigurations/Definition_Updates" already exists - to be managed via Terraform this resource needs to be imported into the State. Please see the resource documentation for "azurerm_maintenance_assignment_virtual_machine" for more information.
│ 
│   with module.virtualmachine["vm1"].azurerm_maintenance_assignment_virtual_machine.definition_updates,
│   on VM/maintenance_assignment.tf line 17, in resource "azurerm_maintenance_assignment_virtual_machine" "definition_updates":
│   17: resource "azurerm_maintenance_assignment_virtual_machine" "definition_updates" {
│ 
╵
##[error]Error: The process '/opt/hostedtoolcache/terraform/1.3.7/x64/terraform' failed with exit code 1
```

### Expected Behaviour

both maintenance configurations should be assigned to the VM

### Actual Behaviour

When we only apply one of the two assignment it works (doesn't matter which assignment we use both are assigned correctly when we use 1 of the two. But when we try to assign both assignment, the deployment will fail (see debug output).

For the record: Via the Azure portal it is possible to assign multiple maintenance configurations to the same virtual machine.

### Steps to Reproduce

```
data "azurerm_maintenance_configuration" "patch_window" {
    name                          = var.maintenance_configuration
    resource_group_name           = data.azurerm_resource_group.resource_group_ka_management.name
  }
    data "azurerm_maintenance_configuration" "definition_updates" {
    name                          = "Definition_Updates"
    resource_group_name           = data.azurerm_resource_group.resource_group_ka_management.name
  }

resource "azurerm_maintenance_assignment_virtual_machine" "patch_window" {
  location                     = azurerm_resource_group.resource_group.location
  maintenance_configuration_id = data.azurerm_maintenance_configuration.patch_window.id
  virtual_machine_id           = azurerm_windows_virtual_machine.virtual_machine.id
}

resource "azurerm_maintenance_assignment_virtual_machine" "definition_updates" {
  location                     = azurerm_resource_group.resource_group.location
  maintenance_configuration_id = data.azurerm_maintenance_configuration.definition_updates.id
  virtual_machine_id           = azurerm_windows_virtual_machine.virtual_machine.id
}
```

**Comment by rob31231232** [comment]
@ziyeqf A few weeks ago you solved #19790 I think this issue has something to do with it. What do you think?

**Comment by ziyeqf** [comment]
Hi @rob31231232 , with checking source code and your config in description, could you please check whether `var.maintenance_configuration` is as same as `Definition_Updates`?

**Comment by rob31231232** [comment]
> 

We set `var.maintenance_configuration` to the name of the Maintenance configuration name.

Some more info: we have multiple maintenance_configrations and each maintenance configuration has a different maintenance window. Lets say that we have 2 maintenance configuration, 1 named "Updates each Wednesday" and 1 "Updates each Thursday". So when we deploy a new VM we set `var.maintenance_configuration` to the maintenance configuration name, for example: `var.maintenance_configuration = "Updates each Wednesday"`.
We also have an maintenance configuration with the name "Definition_Updates". This maintenance configuration has an endless maintenance window because we want the Definition Updates asap.
So each VM will get the mandatory Definition Updates maintenance confirgation, and 1 of the Update maintenance configuration (for example "Updates each Wednesday")
